voltdbclient/distributor.go

```diff
@@ -173,8 +173,9 @@ func (c *Conn) getConn() *nodeConn {
 	if nc.isClosed() {
 		for {
 			n := rand.Intn(size)
-			if n != idx {
-				return c.connected[n]
+			nc = c.connected[n]
+			if !nc.isClosed() {
```

Comment:
Seems an infinity loop if none open connection available, is there a way out of here?

Reply:
No, for one node We aren't even supposed to loop. I will fix that.